### PR TITLE
statistics: fix the hot region API cannot work without hot scheduler

### LINKED ISSUE #3879: hot region should work if hot region schedule is disable
## Bug Report

I find hot region api will not work if hot region schedule is disable 
### What did you do?
schedule remove balance-hot-region-scheduler, See:
https://github.com/tikv/pd/blob/48963369ef04efaa9144eff27135c9ae14429a80/server/cluster/coordinator.go#L444-L456

### What did you expect to see?

hot region api should not be contorlled by  hot region schedule config

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

close #3879. But this PR should be reviewed after the following PRs are merged:

- [x] #4426 
- [x] #4149

### What is changed and how it works?

This PR mainly makes the following changes:

- remove `hasHotStatus` interface
- no need to run a hot scheduler to get hot statistics
- move all calculation logic of hot region into `statistics` package

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: server/statistics/metrics.go [hunk 1]
```diff
@@ -170,6 +170,14 @@ var (
 			Name:      "flow_queue_status",
 			Help:      "Status of the hotspot flow queue.",
 		}, []string{"type"})
+
+	hotPeerSummary = prometheus.NewGaugeVec(
```
Comment (bufferflies) [inline server/statistics/metrics.go:174]: Will it  move all hot perr summary from hot region scheduler to statustucs?

Comment (rleungx) [reply server/statistics/metrics.go:174]: > will the metrics works well if hot regions scheduler is disable.

The metrics don't have any relationship with the hot scheduler. 
> Will it move all hot perr summary from hot region scheduler to statistics?

I don't get it.

**Comment by rleungx** [comment]
The lock problem needs to pay attention to.

File: server/cluster/coordinator.go [hunk 2]
```diff
@@ -535,35 +517,33 @@ func (c *coordinator) resetSchedulerMetrics() {
 
 func (c *coordinator) collectHotSpotMetrics() {
```
Comment (bufferflies) [inline server/cluster/coordinator.go:518]: There is some difference between collection and  GetHotXXXXRegions.

**Review by bufferflies — APPROVED** [review]

**Comment by nolouch** [comment]
ptal @lhy1024

File: server/cluster/cluster.go [hunk 3]
```diff
@@ -1590,23 +1581,17 @@ func (checker *prepareChecker) collect(region *core.RegionInfo) {
 
 // GetHotWriteRegions gets hot write regions' info.
 func (c *RaftCluster) GetHotWriteRegions(storeIDs ...uint64) *statistics.StoreHotPeersInfos {
-	c.RLock()
```
Comment (lhy1024) [inline server/cluster/cluster.go:1593]: Do we no longer need locks?

File: server/statistics/store_hot_peers_infos.go [hunk 4]
```diff
@@ -14,6 +14,19 @@
...
 
+import (
+	"fmt"
+
+	"github.com/tikv/pd/server/core"
+)
+
+const (
+	// HotReadRegionType is hot read region scheduler type.
+	HotReadRegionType = "hot-read-region"
```
Comment (lhy1024) [inline server/statistics/store_hot_peers_infos.go:25]: Where do we use it?

File: server/cluster/cluster.go (hunk 3 above)
Comment (rleungx) [reply server/cluster/cluster.go:1593]: I think so.

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**